addons/mrp/models/product.py

```diff
@@ -109,7 +109,8 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
         This override is used to get the correct quantities of products
         with 'phantom' as BoM type.
         """
-        bom_kits = self.env['mrp.bom']._get_product2bom(self, bom_type='phantom')
+        products = self.filtered(lambda p: p.type  != 'service')
+        bom_kits = {k:v for (k,v) in self.env['mrp.bom']._get_product2bom(products, bom_type='phantom').items() if v}
```

Comment:
you can simplify by `bom_kits = self.env['mrp.bom']._get_product2bom(products, bom_type='phantom')` because of the defaultdict return by the `_get_product2bom`